x-pack/plugins/security_solution/server/lib/detection_engine/rules/get_prepackaged_rules.ts

```diff
@@ -52,7 +69,93 @@ export const validateAllPrepackagedRules = (
   });
 };
 
-export const getPrepackagedRules = (
-  // @ts-expect-error mock data is too loosely typed
-  rules: AddPrepackagedRulesSchema[] = rawRules
-): AddPrepackagedRulesSchemaDecoded[] => validateAllPrepackagedRules(rules);
+const isRuleTemplate = (path: string) => {
+  const pathParts = getPathParts(path);
+  return pathParts.type === 'rules' && pathParts.file !== 'CHANGELOG.json';
+};
+
+// non-blocking check the latest version
+export const getLatestRulesPackageVersion = (): string | undefined => {
+  // kick off a background check for the latest package version.
+  // next time this is called, it might point to the latest.
+  // for on-prem, EPR is not reachable, but we don't want to timeout waiting to connect
+  checkAndStageUpdate();
+
+  // return the most recently known version
+  return latestRulesPackageVersion;
+};
+
+/**
+ * Retrieve a package of rules from the Elastic Package Registry.
+ * @param pkgVersion Specific package version to retrieve from EPR
+ */
+export const getPackageRegistryRules = async (
+  pkgVersion: string
+): Promise<AddPrepackagedRulesSchemaDecoded[]> => {
+  if (pkgVersion === latestRulesPackageVersion) {
+    return latestRulesDownload;
+  }
+
+  const { paths } = await getRegistryPackage(DetectionRulesPackageName, pkgVersion);
+
+  const rulePaths = paths.filter(isRuleTemplate);
+  const rulePromises = rulePaths.map(async (path) => {
+    const content = JSON.parse(getAsset(path).toString('utf8'));
+    return content as AddPrepackagedRulesSchema;
+  });
```

Comment:
I think you can just do this:

```ts
  const rulePromises = rulePaths.map<AddPrepackagedRulesSchema>((path) => {
    return JSON.parse(getAsset(path).toString('utf8'));
  });

  return validateAllPrepackagedRules(rulePromises);
```

Not seeing anything with Promises here going on.

Reply:
4311ef48f1275dd6dc9ee384747287a1d0800ddd